File: storage/db-core/src/main/java/com/coffee/gu/converter/AesConverter.java
package com.coffee.gu.converter;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
public record AesConverter(
        AesHelper aesHelper
) implements AttributeConverter<String, String> {

    @Override
    public String convertToDatabaseColumn(String attribute) {
        if (attribute == null) {
            return null;
        }
        return aesHelper.encrypt(attribute);
    }

    @Override
    public String convertToEntityAttribute(String dbData) {
        if (dbData == null) {
            return null;
        }
        return aesHelper.decrypt(dbData);
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/converter/AesConverter.kt
package com.coffee.gu.converter

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class AesConverter(
    private val aesHelper: AesHelper,
) : AttributeConverter<String, String> {

    override fun convertToDatabaseColumn(attribute: String?): String? {
        if (attribute == null) return null
        return aesHelper.encrypt(attribute)
    }

    override fun convertToEntityAttribute(dbData: String?): String? {
        if (dbData == null) return null
        return aesHelper.decrypt(dbData)
    }
}
